### ISSUE TITLE
[Dialog] The Save changes button has the wrong accessible name and the close button has no accessible name

### ISSUE THREAD

**Issue by rianrietveld** [radix-ui/primitives#1467]
### Current Behavior

The close button has no other content than just the SVG. This means for screen readers this will be announced as "button" with no context of what it does.

In the frontend for the Save Changes button:
`<button aria-label="Close" type="button" class="c-hYXuTU c-hYXuTU-eqKpNk-variant-green">Save changes</button>`
The aria-label overrides the content inside.

In the frontend for the close button:
```
<button type="button" class="c-kLfaJB">
    <svg width="15" height="15" viewBox="0 0 15 15" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path d="M11.7816 4.03157C12.0062 3.80702 12.0062 3.44295 11.7816 3.2184C11.5571 2.99385 11.193 2.99385 10.9685 3.2184L7.50005 6.68682L4.03164 3.2184C3.80708 2.99385 3.44301 2.99385 3.21846 3.2184C2.99391 3.44295 2.99391 3.80702 3.21846 4.03157L6.68688 7.49999L3.21846 10.9684C2.99391 11.193 2.99391 11.557 3.21846 11.7816C3.44301 12.0061 3.80708 12.0061 4.03164 11.7816L7.50005 8.31316L10.9685 11.7816C11.193 12.0061 11.5571 12.0061 11.7816 11.7816C12.0062 11.557 12.0062 11.193 11.7816 10.9684L8.31322 7.49999L11.7816 4.03157Z" fill="currentColor" fill-rule="evenodd" clip-rule="evenodd">
      </path>
   </svg>
</button>
```

### Expected behavior
The "Save changes" button should have no aria-label, the text inside is perfect on it's own and should not be overwritten.
The "Close" button must have an accessible name, so a screen reader can announce the purpose of the button and voice recognition software can target the button more easily.

### Reproducible example

[Dialog on radix-ui.com](https://www.radix-ui.com/docs/primitives/components/dialog)
VoiceOver recording:

https://user-images.githubusercontent.com/2351479/173365056-e67ba2d6-aa60-4526-b38f-86dc526e713d.mov

### Suggested solution

In the frontend for the Save Changes button, remove the aria-label:
`<button type="button" class="c-hYXuTU c-hYXuTU-eqKpNk-variant-green">Save changes</button>`

In the frontend for the close button, add the aria-label "Close" 
```
<button type="button" class="c-kLfaJB" aria-label="close">
    <svg [...]</svg>
</button>
```

Or even better: Add visual text, then also people who don't know what the X means benefit too:
```
<button type="button" class="c-kLfaJB" aria-label="close"> 
     Close
    <svg [...]</svg>
</button>
```

### Your environment

MacOs Monterey 21.1 with Safari 15.2 and VoiceOver.
Tested software on [Dialog on radix-ui.com](https://www.radix-ui.com/docs/primitives/components/dialog)
